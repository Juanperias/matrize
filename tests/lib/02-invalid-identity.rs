use matrize::*;

fn main() {
    let _ = Matrix::<2, 3>::identity();
}
