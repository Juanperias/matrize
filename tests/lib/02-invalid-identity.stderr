error[E0599]: no function or associated item named `identity` found for struct `matrize::Matrix<2, 3>` in the current scope
 --> ./tests/lib/02-invalid-identity.rs:4:29
  |
4 |     let _ = Matrix::<2, 3>::identity();
  |                             ^^^^^^^^ function or associated item not found in `Matrix<2, 3>`
  |
note: if you're trying to build a new `matrize::Matrix<2, 3>` consider using one of the following associated functions:
      matrize::Matrix::<N, M, E>::new
      matrize::Matrix::<N, M, E>::zeros
  --> src/lib.rs
   |
   |     pub const fn new(value: E) -> Matrix<N, M, E> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
   |     pub const fn zeros() -> Matrix<N, M, E> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: the function or associated item was found for
           - `matrize::Matrix<N, N, E>`
